uk.co.compucorp.civicrm.hrleaveandabsences/api/v3/LeaveRequest.php

```diff
@@ -429,3 +429,198 @@ function civicrm_api3_leave_request_ismanagedby($params) {
 
   return $result;
 }
+
+/**
+ * LeaveRequest.addComment API spec
+ *
+ * @param $spec
+ */
+function _civicrm_api3_leave_request_addcomment_spec(&$spec) {
+  $spec['comment_id'] = [
+    'name' => 'comment_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Comment ID',
+    'description' => 'The Comment ID',
+    'api.required' => 0
+  ];
+
+  $spec['leave_request_id'] = [
+    'name' => 'leave_request_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Leave Request ID',
+    'description' => 'The ID of the Leave Request',
+    'api.required' => 1,
+  ];
+
+  $spec['text'] = [
+    'name' => 'text',
+    'type' => CRM_Utils_Type::T_STRING,
+    'title' => 'Text',
+    'description' => 'The comment text',
+    'api.required' => 1,
+  ];
+
+  $spec['contact_id'] = [
+    'name' => 'contact_id',
+    'type' => CRM_Utils_Type::T_INT,
+    'title' => 'Contact (Commenter)',
+    'description' => 'The contact who made the comment',
+    'api.required' => 1,
+    'FKClassName'  => 'CRM_Contact_DAO_Contact',
+    'FKApiName'    => 'Contact',
+  ];
+}
+
+/**
+ * LeaveRequest.addComment API
+ *
+ * Uses the API exposed by the Comment entity (Comment.create)
+ * to create comments related to a LeaveRequest.
+ *
+ * @param $params
+ *
+ * @return array
+ */
+function civicrm_api3_leave_request_addcomment($params) {
+  _civicrm_api3_leave_request_prepare_parameters_for_comment_payload($params);
+  $results =  civicrm_api3('Comment', 'create', $params);
```

Comment:
I thought this was going to be implemented on the LeaveRequest Service. Something like:

```php
public function addComment($leaveRequestID, $text, $contactID) {
  // call Comment API 
 // filter and return data
}
```

And then, on the API function:
```php
function civicrm_api3_leave_request_addcomment($params) {
  $servive = new LeaveRequestService();
  $service->addComment($params['leave_request_id'], $params['contact_id'], $params['text'])
}
```